firefox-ios/Client/Application/WindowManager.swift

```diff
@@ -208,12 +209,25 @@ final class WindowManagerImplementation: WindowManager, WindowTabsSyncCoordinato
         let result: ReservedWindowUUID
         // TODO: [9610] Unit tests should eventually be updated for these changes. Forthcoming.
         if !isIpad && !AppConstants.isRunningUnitTest {
-            // At this point we should always have either a single UUID on disk or
-            // no UUIDs because this is a brand new app install
+            // We should always have either a single UUID on disk or no UUIDs because this is a brand new app install
             if onDiskUUIDs.isEmpty {
                 result = ReservedWindowUUID(uuid: WindowUUID(), isNew: true)
             } else {
                 result = ReservedWindowUUID(uuid: onDiskUUIDs.first!, isNew: false)
+
+                let uuidCount = onDiskUUIDs.count
+                if uuidCount > 1 {
+                    // This is unexpected. Potentially related to incident in v128 (see: FXIOS-9533).
+                    // On iPhone, we should never have more than 1 window tab file. Log an error and
+                    // clean up the UUID(s) that we know won't be used. We expect a certain number of
+                    // these fatal errors to be logged for users previously impacted by the above, and
+                    // then it should fall to zero.
+                    logger.log("Detected multiple window tab files on iPhone (UUID count: \(uuidCount))",
```

Comment:
@OrlaM @adudenamedruby There is a possibility here that in some cases this async deletion task might not complete before the app is terminated (e.g. if the app is killed almost immediately after it's launched). In this case, there's a chance we might see this fatal log more than once for a user.

However that should be _exceedingly_ rare given how fast this is going to be performed during the app startup. So it should still be close to a 1:1 event-to-user ratio, and in general we should not see the error repeated.

If anyone has concerns or suggestions though LMK.

Reply:
Personally, I think it's probably fine. As long as we're in the 1:1 ballpark, we should be good.

if we wanted to be 100% certain that we only send this error once, we could add a temporary thing that we've already sent this error in user defaults, and get rid of it in a few releases, while keeping the async delete untouched so it can do it, if it wasn't completed. I dislike this solution, but not sure what else would guarantee us sending this error only once.